File: device/src/HttpSubsystem/Message.cpp
#include <HttpSubsystem/Message.h>

Message::Message(JsonDocument &doc)
{
    id = doc["id"];
    content = doc["content"].as<String>();
    ack = doc["ack"].as<bool>();
    JsonArray jsonLeds = doc["leds"];
    for (int idx = 0; idx < 64; idx++)
    {
        leds[idx] = jsonLeds[idx].as<bool>();
    }
}

Message::Message() {}

File: application/Models/Message.cs
namespace matrix_web.Models;

public class Message {
    [LiteDB.BsonId]
    public long id { get; set; }

    public DateTime date { get; set; }

    public String content { get; set; }

    public bool[] leds { get; set; }

    public bool ack { get; set; }

    public Message(long id, DateTime date, String content, bool[] leds, bool ack) {
        this.id = id;
        this.date = date;
        this.content = content;
        this.leds = leds;
        this.ack = ack;
    }

    public Message() {
        date = DateTime.Now;
        content = "";
        leds = new bool[0];
     }
}